serialization.md

```diff
@@ -574,15 +595,27 @@ Merkle Tree Aux Functions
     rule HPEncodeAux ( X ) => 0 requires         X ==Int 0
     rule HPEncodeAux ( X ) => 2 requires notBool X ==Int 0
 
+    syntax Map ::= #cleanBranchMap( Map )              [function]
+                 | #cleanBranchMapAux( Map, Set, Set ) [function]
+ // -------------------------------------------------------------
+    rule #cleanBranchMap( M ) => #cleanBranchMapAux( M, keys(M), .Set )
+
+    rule #cleanBranchMapAux(                   M,                      .Set,                      S ) => removeAll( M, S )
+    rule #cleanBranchMapAux( X |-> .MerkleTree _, (SetItem(X) => .Set) _   , (.Set => SetItem(X)) _ )
+    rule #cleanBranchMapAux(                   _, (SetItem(X) => .Set) _   ,                      _ ) [owise]
```

Comment:
Use a `List` here instead of a `Set`. You can use `keys_list(M)` instead of `keys(M)` to automatically get a `List` out.

The pattern-matching on arbitrary elements of a `Set` is non-deterministic, which can be slow, whereas with a `List` you can always just take the first element.